modules/mono/godotsharp_dirs.cpp

```diff
@@ -139,18 +171,37 @@ class _GodotSharpDirs {
 #endif
 		api_assemblies_dir = api_assemblies_base_dir.path_join(GDMono::get_expected_api_build_config());
 #else // TOOLS_ENABLED
+		String platform = _get_platform_name();
 		String arch = Engine::get_singleton()->get_architecture_name();
 		String appname_safe = path::get_csharp_project_name();
-		String data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + arch);
+		String data_dir_root = exe_dir.path_join("data_" + appname_safe + "_" + platform + "_" + arch);
 		if (!DirAccess::exists(data_dir_root)) {
-			data_dir_root = exe_dir.path_join("data_Godot_" + arch);
+			data_dir_root = exe_dir.path_join("data_Godot_" + platform + "_" + arch);
```

Comment:
Can we get rid of the fallback to `data_Godot`? It doesn't seem like our exporter ever generates a directory with that name.

Reply:
Yeah I think we can remove it.

BTW, do we warn users well enough if the data dir is missing?

Reply:
Not really, it seems we just crash when we inevitably fail to load hostfxr. In the terminal you'll at least get to see `ERROR: .NET: Failed to load hostfxr` but it's not very user-friendly since it doesn't explain why it couldn't find it.